### ISSUE TITLE
Not the same environment for a debug and a non-debug build

### ISSUE THREAD

**Issue by johannesring** [hashdist/hashdist#360]
I just noticed that the environment is different when running a standard `hit build` and when running `hit build --debug`. This is because HashDist run the following command to start the debug shell:

``` console
/bin/bash --noprofile --rcfile <rcfile>
```

The problem with this is that it will inherit the environment from the parent process. One fix I have found is to run the same command with `env -i`, like this:

``` console
env -i /bin/bash --noprofile --rcfile <rcfile>
```

The `-i` option will make sure that the command is started with an empty environment. Below is the patch. I can make a pull request if this sounds okay.

``` patch
diff --git a/hashdist/core/run_job.py b/hashdist/core/run_job.py
index 2361da3..20e2276 100644
--- a/hashdist/core/run_job.py
+++ b/hashdist/core/run_job.py
@@ -679,7 +679,7 @@ class CommandTreeExecution(object):
                 sys.stderr.write('  %s\n' % args)
                 sys.stderr.write('\n')
                 sys.stderr.write('When you are done, "exit 1" to abort build, or "exit 0" to continue.\n\n')
-                proc = subprocess.Popen([self.debug_shell, '--noprofile', '--rcfile', rcfile])
+                proc = subprocess.Popen(['env', '-i', self.debug_shell, '--noprofile', '--rcfile', rcfile])
                 retcode = proc.wait()
                 if retcode != 0:
                     self.logger.error("Debug build manually aborted")
```

**Comment by dagss** [comment]
Looks good to me, assuming you tested it on both linux and osx (in which case just merge to master...)

**Comment by johannesring** [comment]
Yes, it works on both linux and osx.

**Comment by johannesring** [comment]
Fixed in e1be790.
